src/altair/block_processing.rs

```diff
@@ -29,9 +46,123 @@ pub fn process_attestation<
         SYNC_COMMITTEE_SIZE,
     >,
-    _attestation: &Attestation<MAX_VALIDATORS_PER_COMMITTEE>,
-    _context: &Context,
+    attestation: &Attestation<MAX_VALIDATORS_PER_COMMITTEE>,
+    context: &Context,
 ) -> Result<()> {
+    let data = &attestation.data;
+
+    let is_previous = data.target.epoch == get_previous_epoch(state, context);
+    let current_epoch = get_current_epoch(state, context);
+    let is_current = data.target.epoch == current_epoch;
+    let valid_target_epoch = is_previous || is_current;
+    if !valid_target_epoch {
+        return Err(invalid_operation_error(InvalidOperation::Attestation(
+            InvalidAttestation::InvalidTargetEpoch {
+                target: data.target.epoch,
+                current: current_epoch,
+            },
+        )));
+    }
+
+    let attestation_epoch = compute_epoch_at_slot(data.slot, context);
+    if data.target.epoch != attestation_epoch {
+        return Err(invalid_operation_error(InvalidOperation::Attestation(
+            InvalidAttestation::InvalidSlot {
+                slot: data.slot,
+                epoch: attestation_epoch,
+                target: data.target.epoch,
+            },
+        )));
+    }
+
+    let attestation_has_delay = data.slot + context.min_attestation_inclusion_delay <= state.slot;
+    let attestation_is_recent = state.slot <= data.slot + context.slots_per_epoch;
+    let attestation_is_timely = attestation_has_delay && attestation_is_recent;
+    if !attestation_is_timely {
+        return Err(invalid_operation_error(InvalidOperation::Attestation(
+            InvalidAttestation::NotTimely {
+                state_slot: state.slot,
+                attestation_slot: data.slot,
+                lower_bound: data.slot + context.slots_per_epoch,
+                upper_bound: data.slot + context.min_attestation_inclusion_delay,
+            },
+        )));
+    }
+
+    let committee_count = get_committee_count_per_slot(state, data.target.epoch, context);
+    if data.index >= committee_count {
+        return Err(invalid_operation_error(InvalidOperation::Attestation(
+            InvalidAttestation::InvalidIndex {
+                index: data.index,
+                upper_bound: committee_count,
+            },
+        )));
+    }
+
+    let committee = get_beacon_committee(state, data.slot, data.index, context)?;
+    if attestation.aggregation_bits.len() != committee.len() {
+        return Err(invalid_operation_error(InvalidOperation::Attestation(
+            InvalidAttestation::Bitfield {
+                expected_length: committee.len(),
+                length: attestation.aggregation_bits.len(),
+            },
+        )));
+    }
+
+    // Participation flag indices
+    let inclusion_delay = state.slot - data.slot;
+    let participation_flag_indices =
+        get_attestation_participation_flag_indices(state, data, inclusion_delay, context)?;
+
+    // Verify signature
+    let _ = is_valid_indexed_attestation(
```

Comment:
```suggestion
    is_valid_indexed_attestation(
```

Reply:
this is kind of weird and really should prolly return a boolean given how it reads but not super important imo